js/intro.js

```diff
@@ -0,0 +1,305 @@
+    type: 'p',
+    classList: ['user-input', 'input-big'],
+    id: 'renderWhyRds',
+  });
+  renderWhyRds.innerHTML = users.whyRds;
+  container.appendChild(whyRds);
+  container.appendChild(renderWhyRds);
+
+  const heardAbout = createElement({
+    type: 'p',
+    classList: ['input-label-dark'],
+  });
+  heardAbout.innerText = 'How did you hear about us';
+  const renderHeardAbout = createElement({
+    type: 'p',
+    classList: ['user-input', 'input-regular'],
+    id: 'renderHeardAbout',
+  });
+  renderHeardAbout.innerText = users.foundFrom;
+  container.appendChild(heardAbout);
+  container.appendChild(renderHeardAbout);
+  loading.classList.add('hidden');
+  document.querySelector('.intro-main').appendChild(renderIntroPage);
+}
+
+//making userSavedData object from API
+async function showSavedDetails() {
+  try {
+    const userId = urlParams.get('id');
+    const usersRequest = await makeApiCall(`${BASE_URL}/users/${userId}/intro`);
+    if (usersRequest.status === 200) {
+      const userData = usersRequest.data[0];
+      let userSavedData = {
+        firstName: userData.biodata.firstName,
+        lastName: userData.biodata.lastName,
+        city: userData.location.city,
+        state: userData.location.state,
+        country: userData.location.country,
+        introduction: userData.intro.introduction,
+        skills: userData.professional.skills,
+        institution: userData.professional.institution,
+        funFact: userData.intro.funFact,
+        forFun: userData.intro.forFun,
+        whyRds: userData.intro.whyRds,
+        foundFrom: userData.foundFrom,
+      };
+      generateSavedDetailsForm(userSavedData);
+    } else if (usersRequest.status === 404) {
+      generateNoDataFoundPage();
+      loading.classList.add('hidden');
+      setTimeout(() => {
+        alert('SuperUser You Write Wrong userId');
+        location.href = 'https://www.realdevsquad.com/intro.html';
+      }, 1500);
+    }
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+(async function setAuth() {
+  try {
+    const res = await fetch(`${BASE_URL}/users/self`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+
+    const selfDetails = await res.json();
+    if (selfDetails.roles.super_user) {
+      notAuthorized.classList.add('hidden');
+      mainContainer.classList.remove('hidden');
+      if (!urlParams.has('id')) {
+        queryParamsNotValid();
+      } else {
+        showSavedDetails();
+      }
+    } else {
+      if (!urlParams.has('id')) {
+        queryParamsNotValid();
+      } else {
+        notFound.classList.add('hidden');
+        loading.classList.add('hidden');
+        generatenotAuthorizedPage();
+        mainContainer.classList.add('hidden');
+      }
+    }
+  } catch (err) {
+    window.addEventListener('click', showToast('something went wrong', 6000));
+    setTimeout(function () {
+      window.location.href = HOME_URL;
```

Comment:
why are we redirecting the user without informing?

Reply:
i inform user using toast